hack/setup.sh

```diff
@@ -248,49 +354,296 @@ install_tekton() {
                 kapp deploy --yes -a tekton -f-
 }
 
-setup_example() {
-        ytt --ignore-unknown-comments \
-                -f "$DIR/../examples/source-to-knative-service" \
+install_tekton_git_cli_task() {
+  kapp deploy --yes -a tekton-git-cli -f https://raw.githubusercontent.com/tektoncd/catalog/main/task/git-cli/0.2/git-cli.yaml
+}
+
+setup_example_sc() {
+        export test_name="$1"
+        kapp deploy --yes -a "setup-example-$test_name" \
+            -f <(ytt --ignore-unknown-comments \
+                -f "$DIR/../examples/shared" \
+                -f "$DIR/../examples/$test_name/values.yaml" \
                 --data-value registry.server="$REGISTRY" \
                 --data-value registry.username=admin \
                 --data-value registry.password=admin \
-                --data-value image_prefix="$REGISTRY/example-" |
-                kapp deploy --yes -a example -f-
+                --data-value image_prefix="$REGISTRY/example-$test_name-")
+        kapp deploy --yes -a "example-$test_name" \
+            -f <(ytt --ignore-unknown-comments \
+                -f "$DIR/../examples/$test_name" \
+                --data-value registry.server="$REGISTRY" \
+                --data-value registry.username=admin \
+                --data-value registry.password=admin \
+                --data-value workload_name="$test_name" \
+                --data-value image_prefix="$REGISTRY/example-$test_name-")
 }
 
-teardown_example() {
-        kapp delete --yes -a example
+teardown_example_sc() {
+        export test_name="$1"
+        kapp delete --yes -a "example-$test_name"
+        kapp delete --yes -a "setup-example-$test_name"
+        until [[ -z $(kubectl get pods -l "serving.knative.dev/configuration=$test_name" -o name) ]]; do sleep 1; done
```

Comment:
This line has been hanging for me because that pod is not getting deleted and it is stopping the tests from progressing. Is this something wrong on my end (I don't think it is because I don't think the deletes above should remove this pod but I could definitely be wrong)?

Reply:
This path is exercised in the e2e test. I'm not sure why it isn't working on your machine...

Reply:
This succeeds for me on both my personal machine and on a linux VM. But it fails in the github runner. I've commented it out for now.